### PR TITLE
add `ldb unsafe_remove_sst_file` subcommand

### PR DESCRIPTION
**Pull request by ajkr**
This is adapted from #6678 but takes a different approach, avoiding opening a read-write DB and avoiding the `DeleteFile()` API.

First, this PR refactors how options variables are initialized in `ldb` so it can be reused in a subcommand that doesn't open a DB:

- Separated remaining option initialization logic out of `OpenDB()`. The new `PrepareOptions()` function initializes the full options state.
- Fixed an old TODO about applying the subcommand CF option overrides to the proper `ColumnFamilyOptions` object.

Second, this PR adds the `ldb unsafe_remove_sst_file` subcommand. It uses the `VersionSet`-level APIs to remove the file with the specified number.

Test Plan: played with interactive python and this file removal command. Verified openability/correct results in case of multiple column families, multiple levels, etc.

### CODE REVIEW

File: tools/ldb_cmd.cc [hunk 1]
```diff
@@ -3440,8 +3451,79 @@ void ListFileRangeDeletesCommand::DoCommand() {
...
+    options_.db_paths.emplace_back(db_path_, 0 /* target_size */);
+  }
+
+  WriteController wc(options_.delayed_write_rate);
+  WriteBufferManager wb(options_.db_write_buffer_size);
+  ImmutableDBOptions immutable_db_options(options_);
+  std::shared_ptr<Cache> tc(
+      NewLRUCache(1 << 20 /* capacity */, options_.table_cache_numshardbits));
+  EnvOptions sopt;
+  VersionSet versions(db_path_, &immutable_db_options, sopt, tc.get(), &wb, &wc,
```
Comment (Connor1996) [inline tools/ldb_cmd.cc:3498]: does it ensure there is no other DB instance is running?

Comment (ajkr) [reply tools/ldb_cmd.cc:3498]: Good point, it does not.

File: tools/ldb_cmd.cc [hunk 2]
```diff
@@ -3440,8 +3451,79 @@ void ListFileRangeDeletesCommand::DoCommand() {
...
+void UnsafeRemoveSstFileCommand::DoCommand() {
+  PrepareOptions();
+
+  if (options_.db_paths.empty()) {
+    // `VersionSet` expects options that have been through `SanitizeOptions()`,
+    // which would sanitize an empty `db_paths`.
+    options_.db_paths.emplace_back(db_path_, 0 /* target_size */);
+  }
+
+  WriteController wc(options_.delayed_write_rate);
```
Comment (pdillinger) [inline tools/ldb_cmd.cc:3492]: Can you drop a comment in here about why we don't want to open the DB for this?

File: tools/ldb_cmd.cc (hunk 1 above)
Comment (pdillinger) [reply tools/ldb_cmd.cc:3498]: Doesn't scare me too much for a command called "unsafe remove ..." ;)

**Review by ajkr — COMMENTED** [review]
Thanks for the review!

File: tools/ldb_cmd.cc (hunk 2 above)
Comment (ajkr) [reply tools/ldb_cmd.cc:3500]: Done, at the top of this function.

**Pull request closed without merging**